tutorials/continuous_aggs_tutorial.md

```diff
@@ -0,0 +1,171 @@
+
+```sql
+create view cagg_rides_view  WITH 
+(timescaledb.continuous, timescaledb.refresh_interval = ’30m’)
+AS
+select vendor_id, time_bucket('1h', pickup_datetime) as day,
+     count(*) total_rides, 
+     avg(fare_amount) avg_fare, 
+     max(trip_distance) as max_trip_distance,
+     min(trip_distance) as min_trip_distance
+from rides
+group by vendor_id, time_bucket('1h', pickup_datetime);
+```
+Note that a continuous aggregate query specifies a group by clause with a  time_bucket expression and the time_bucket expression uses the time dimension column of the *rides* hypertable.
+```sql
+\d cagg_rides_view
+                          View "public.cagg_rides_view"
+      Column       |            Type             | Collation | Nullable | Default 
+-------------------+-----------------------------+-----------+----------+---------
+ vendor_id         | text                        |           |          | 
+ day               | timestamp without time zone |           |          | 
+ total_rides       | bigint                      |           |          | 
+ avg_fare          | numeric                     |           |          | 
+ max_trip_distance | numeric                     |           |          | 
+ min_trip_distance | numeric                     |           |          | 
+```
+
+We can view the metadata for the continuous aggregate in the **timescaledb_information.continuous_aggregates** view.
+
+``` sql
+select view_name, refresh_lag, refresh_interval, max_interval_per_job, materialization_hypertable from timescaledb_information.continuous_aggregates;
+-[ RECORD 1 ]--------------+-------------------------------------------------
+view_name                  | cagg_rides_view
+refresh_lag                | 02:00:00
+refresh_interval           | 00:30:00
+max_interval_per_job       | 20:00:00
+materialization_hypertable | _timescaledb_internal._materialized_hypertable_2
+```
+The refresh_interval is set to 30 minutes. The computed aggregates are saved in the materialization table,  _timescaledb_internal._materialized_hypertable_2. 
+
+What is refresh_lag and max_interval_per_job? We use the **refresh_lag** to indicate by how much does the continuous aggregate query lag behind the data in the *rides* table. If we expect frequent updates to the rides table for the current hour, we do not want to precompute the aggregates for that range. We set the** refresh_lag = ‘1h’ **to indicate that. (The default is twice the bucket_width used by the time_bucket expression. This is the 2 hours shown for refresh_lag above). So the continuous aggregate will get refreshed every 30 minutes (refresh_interval) but will update the continuous aggregates only for the data that satisfies the condition: time_bucket(‘1h’, pickup_datetime) <   max(pickup_time) - ‘1h’  (as the refresh_lag is set to 1 hour)
```

Comment:
`(as the refresh_lag is set to 1 hour)` but the refresh_lag is 2 hours right? Also I think the formula is actually for the end of the bucket not the beginning. @JLockerman can you help with the wording here?

Reply:
this is for the example case. so the lag is 1 hour. Modified wording as this was not clear.